### ISSUE TITLE
[TextField] clearButton does not have a focus indicator for keyboard-only users

### ISSUE THREAD

**Issue by braheem** [Shopify/polaris-react#3283]
## Issue summary

When tabbing through a TextField component, the user can hit the clearButton but not be aware of it visually. This is an issue for keyboard-only users. 

## Expected behavior

There should be a focus ring or similar to indicate that the clearButton has received focus and is actionable. This should also be the case for readOnly TextFields, since the clearButton is still actionable.

## Actual behavior

When the clearButton is focused with tab, no visual indication is provided. This can be reproduced in a sandbox by going to the official docs https://polaris.shopify.com/components/forms/text-field and selecting example "Text Field with clearButton" and opening a sandbox for it. 

## Steps to reproduce the problem

1. tab through a TextField with a clearButton and notice no visual indicator when the clearButton is focused

## Specifications

- Are you using the React components? (Y/N): Y
- Polaris version number: Latest
- Browser: Chrome
- Device: Macbook
- Operating System:

**Comment by alex-page** [comment]
It seems to have a focus now. Closing this issue. Thanks!
